src/Installers/Windows/AspNetCoreModule-Setup/CustomAction/aspnetcoreCA.cpp

```diff
@@ -0,0 +1,181 @@
+                          (VOID **)&pAdminMgr);
+    if (FAILED(hr))
+    {
+        goto exit;
+    }
+
+    hr = pAdminMgr->GetAdminSection(L"system.webServer/httpCompression",
+                                    L"MACHINE/WEBROOT/APPHOST",
+                                    &pHttpCompressionSection);
+    if (FAILED(hr))
+    {
+        goto exit;
+    }
+
+    hr = pHttpCompressionSection->GetElementByName(L"dynamicTypes",
+                                             &pDynamicCompressionElement);
+    if (FAILED(hr))
+    {
+        goto exit;
+    }
+
+    hr = pDynamicCompressionElement->get_Collection(&pMimeTypeCollection);
+    if (FAILED(hr))
+    {
+        goto exit;
+    }
+
+    hr = pMimeTypeCollection->get_Count(&i);
+    if (FAILED(hr) || i == 0)
+    {
+        // failure or DynamicCmpression is not enabled
+        goto exit;
+    }
+
+    for (i=0; i<_countof(gMimeTypes); i++)
+    {
+        hr = pMimeTypeCollection->CreateNewElement(L"add",
+                                                   &pMimeTypeElement);
+        if (FAILED(hr))
+        {
+            goto exit;
+        }
+
+        hr = VariantAssign(&varName,
+                           gMimeTypes[i].pszMimeType);
+        if (FAILED(hr))
+        {
+            goto exit;
+        }
+
+        hr = SetElementProperty(pMimeTypeElement,
+                                L"mimeType",
+                                &varName);
+        if (FAILED(hr))
+        {
+            goto exit;
+        }
+        VariantClear(&varName);
+
+        varName.vt = VT_BOOL;
+        varName.boolVal = gMimeTypes[i].fEnabled ? VARIANT_TRUE : VARIANT_FALSE;
+
+        hr = SetElementProperty(pMimeTypeElement,
+                                L"enabled",
+                                &varName);
+        if (FAILED(hr))
+        {
+            goto exit;
+        }
+        VariantClear(&varName);
+
+        hr = pMimeTypeCollection->AddElement(pMimeTypeElement);
+        if (FAILED(hr) &&
+            hr != HRESULT_FROM_WIN32(ERROR_ALREADY_EXISTS))
+        {
+            goto exit;
+        }
+
+        pMimeTypeElement->Release();
+        pMimeTypeElement = NULL;
+    }
+
+    hr = pAdminMgr->CommitChanges();
+
+ exit:
+
+    VariantClear(&varName);
+
+    if (pMimeTypeElement != NULL)
+    {
+        pMimeTypeElement->Release();
+        pMimeTypeElement = NULL;
+    }
+
+    if (pMimeTypeCollection != NULL)
+    {
+        pMimeTypeCollection->Release();
+        pMimeTypeCollection = NULL;
+    }
+
+    if (pDynamicCompressionElement != NULL)
+    {
+        pDynamicCompressionElement->Release();
+        pDynamicCompressionElement = NULL;
+    }
+
+    if (pHttpCompressionSection != NULL)
+    {
+        pHttpCompressionSection->Release();
+        pHttpCompressionSection = NULL;
+    }
+
+    if (pAdminMgr != NULL)
+    {
+        pAdminMgr->Release();
+        pAdminMgr = NULL;
+    }
+
+    return (SUCCEEDED(hr)) ? ERROR_SUCCESS : ERROR_SUCCESS;
```

Comment:
Is returning `ERROR_SUCCESS` on failure desired here?

Reply:
It isn't desired, but we end up ignoring the return value no matter what: https://github.com/aspnet/Universe/blob/768c91cb75e76ad804387353861aaaa313e878f5/src/Installers/Windows/AspNetCoreModule-Setup/Installers/ANCMV2/aspnetcoremodulev2.wxs#L290